MSAL/src/instance/MSALAdfsAuthorityResolver.m

```diff
@@ -207,15 +211,18 @@ - (void)webFingerRequestForEndpoint:(NSString *)authenticationEndpoint
     CHECK_ERROR_COMPLETION(authenticationEndpoint, context, MSALErrorInvalidParameter, @"AuthenticationEndpoint cannot be nil.");
     CHECK_ERROR_COMPLETION(authority, context, MSALErrorInvalidParameter, @"authority cannot be nil.");
     
-    MSALWebAuthRequest *request =
-    [[MSALWebAuthRequest alloc] initWithURL:[MSALAdfsAuthorityResolver urlForWebFinger:authenticationEndpoint absoluteAuthority:authority.absoluteString]
-                                    context:context];
-    
-    [request sendGet:^(MSALHttpResponse *response, NSError *error) {
+    NSURL *url = [MSALAdfsAuthorityResolver urlForWebFinger:authenticationEndpoint absoluteAuthority:authority.absoluteString];
+    MSIDAADAuthorityValidationRequest *request = [[MSIDAADAuthorityValidationRequest alloc] initWithUrl:url
+                                                                                                context:context];
+    [request sendWithBlock:^(id response, NSError *error) {
+        
+        [request finishAndInvalidate];
+        
         CHECK_COMPLETION(!error);
         
+        NSMutableDictionary *responseDic = (NSMutableDictionary *)response;
```

Comment:
nit: should we check that response is actually a dictionary? or should the request maybe already return a dictionary?

Reply:
done